snapcraft/plugins/catkin.py

```diff
@@ -312,3 +326,102 @@ def _handle_package(self, pkg):
         ])
 
         self._rosrun(catkincmd)
+
+
+class _Rosdep:
+    def __init__(self, ros_distro, ros_package_path, rosdep_path,
+                 ubuntu_sources):
+        self._ros_distro = ros_distro
+        self._ros_package_path = ros_package_path
+        self._ubuntu_sources = ubuntu_sources
+        self._rosdep_path = rosdep_path
+        self._rosdep_install_path = os.path.join(self._rosdep_path, 'install')
+        self._rosdep_sources_path = os.path.join(self._rosdep_path,
+                                                 'sources.list.d')
+        self._rosdep_cache_path = os.path.join(self._rosdep_path, 'cache')
+
+    def setup(self):
+        os.makedirs(self._rosdep_install_path, exist_ok=True)
+        os.makedirs(self._rosdep_sources_path, exist_ok=True)
+        os.makedirs(self._rosdep_cache_path, exist_ok=True)
+
+        # Prepare to download Ubuntu .deb packages from the ROS repositories.
+        # We need to do this here because the package dependencies are not
+        # necessarily known at pull-time-- we need to interrogate the Catkin
+        # packages first. Then we pull them down ourselves.
+        ubuntu = repo.Ubuntu(self._rosdep_path, sources=self._ubuntu_sources)
+
+        # rosdep is used for analyzing the dependencies specified within each
+        # project's package.xml. But it introduces a complication, in that it's
+        # not a dependency of the project, and we don't want to bloat the .snap
+        # more than necessary. So we'll unpack it somewhere else, and use it
+        # from there.
+        logger.info("Fetching rosdep...")
```

Comment:
Ah, no thanks! I've been trying to adhere but I knew I was going to miss some, haha :) .